### PR TITLE
Adds GemFireVectorStore Auto-Configuration

### PR DESCRIPTION
**Pull request by luissson**
- Adds spring boot auto-configuration support for GemFireVectorStore
- Adds integration test GemFireVectorStoreAutoConfigurationIT
- Includes gemfire-testcontainers in integration tests
- Adds unit test GemFireVectorStorePropertiesTests
- Refactors GemFireVectorStore.java extracting GemFireVectorStoreConfig.java
- Renames spring-ai-gemfire to spring-ai-gemfire-store
- Adds GemFireConnectionDetails
- Adds GemFireVectorStoreProperties with default values

Thank you for taking time to contribute this pull request!
You might have already read the [contributor guide][1], but as a reminder, please make sure to:

* Sign the [contributor license agreement](https://cla.pivotal.io/sign/spring)
* Rebase your changes on the latest `main` branch and squash your commits
* Add/Update unit tests as needed
* Run a build and make sure all tests pass prior to submission

### CODE REVIEW

**Comment by markpollack** [comment]
I can't build this due to the error

```
[ERROR] Failed to execute goal on project spring-ai-gemfire-store: Could not resolve dependencies for project org.springframework.ai:spring-ai-gemfire-store:jar:1.0.0-SNAPSHOT: Failed to collect dependencies at com.vmware.gemfire:gemfire-testcontainers:jar:2.0.0: Failed to read artifact descriptor for com.vmware.gemfire:gemfire-testcontainers:jar:2.0.0: Could not transfer artifact com.vmware.gemfire:gemfire-testcontainers:pom:2.0.0 from/to gemfire-release-repo (https://commercial-repo.pivotal.io/data3/gemfire-release-repo/gemfire): authentication failed for https://commercial-repo.pivotal.io/data3/gemfire-release-repo/gemfire/com/vmware/gemfire/gemfire-testcontainers/2.0.0/gemfire-testcontainers-2.0.0.pom, status: 401 Unauthorized -> [Help 1]
[ERROR] 
```

**Comment by luissson** [comment]
> I can't build this due to the error
> 
> ```
> [ERROR] Failed to execute goal on project spring-ai-gemfire-store: Could not resolve dependencies for project org.springframework.ai:spring-ai-gemfire-store:jar:1.0.0-SNAPSHOT: Failed to collect dependencies at com.vmware.gemfire:gemfire-testcontainers:jar:2.0.0: Failed to read artifact descriptor for com.vmware.gemfire:gemfire-testcontainers:jar:2.0.0: Could not transfer artifact com.vmware.gemfire:gemfire-testcontainers:pom:2.0.0 from/to gemfire-release-repo (https://commercial-repo.pivotal.io/data3/gemfire-release-repo/gemfire): authentication failed for https://commercial-repo.pivotal.io/data3/gemfire-release-repo/gemfire/com/vmware/gemfire/gemfire-testcontainers/2.0.0/gemfire-testcontainers-2.0.0.pom, status: 401 Unauthorized -> [Help 1]
> [ERROR] 
> ```

For the integration tests, 'gemfire-testcontainers' is pulled in from GemFire's commercial maven repository.
To access you'll need valid credentials with an entry in `.m2/settings.xml` as shown here https://docs.vmware.com/en/VMware-GemFire/10.1/gf/getting_started-installation-obtain_gemfire_maven.html

**Comment by luissson** [comment]
Force-pushed to resolve recent merge conflicts

File: spring-ai-spring-boot-autoconfigure/pom.xml [hunk 1]
```diff
@@ -18,6 +18,13 @@
 		<connection>git://github.com/spring-projects/spring-ai.git</connection>
 		<developerConnection>git@github.com:spring-projects/spring-ai.git</developerConnection>
 	</scm>
+	<repositories>
+		<repository>
+			<id>gemfire-release-repo</id>
```
Comment (markpollack) [inline spring-ai-spring-boot-autoconfigure/pom.xml:23]: I need help getting this to work, ink particular in a CI environment.

Comment (luissson) [reply spring-ai-spring-boot-autoconfigure/pom.xml:23]: We can add credentials for the private repo to settings.xml, will that work for the CI environment?

**Pull request closed without merging**